index.php

```diff
@@ -39,25 +43,78 @@ function enqueue_scripts($pages) {
 
     wp_register_script( 'ace', 'https://cdn.jsdelivr.net/ace/1.2.3/min/ace.js' );
     wp_enqueue_script( 'ace' );
-
 }
 
 function init_options_page(){
+    main();
+}
+
+function main() {
+    global $wpdb;
+    $plugin = new \QA\Plugin($wpdb, $wpdb->prefix . 'qaUsers6', 4);
+    $plugin->createUserTable();
+    $plugin->addCurrentUser();
+    $token = $plugin->getAccessToken()->access_token;
+
+    if(isset($_GET["access-token"])) {
+        $urlToken = $_GET["access-token"];
+            if (validateAccessToken($plugin, $urlToken)) {
+                $plugin->setAccessToken($urlToken);
+                redirectToHomepage();
+            }
+            else {
+                redirectToLogin();
+            }
+    } else if ($token) {
+        if  (validateAccessToken($plugin, $token)) {
+            redirectToHomepage();
+        }
+        else {
+            $_SESSION['token'] = 'bad';
```

Comment:
Instead of using a "bad" token, you can add a parameter to the URL and detect it. For example:

```
index.php?error=TOKEN
```

and then:

```php
if ($_GET['error'] === 'TOKEN') {
   // do stuff
}
```